app/controllers/idv_controller.rb

```diff
@@ -32,6 +33,22 @@ def activated
 
   private
 
+  def clear_session
+    user_session['idv/doc_auth'] = {}
+    user_session['idv/in_person'] = {}
+    user_session['idv/inherited_proofing'] = {}
+    idv_session.clear
+    Pii::Cacher.new(current_user, user_session).delete
+  end
+
+  def enrollment
+    InPersonEnrollment.where(user_id: current_user.id).last
+  end
+
+  def enrollment_status
+    enrollment.present? ? enrollment.status : ''
+  end
```

Comment:
We can minimize the query footprint like so:

```suggestion
  def enrollment_status
    InPersonEnrollment.where(user_id: 6).order(id: :desc).pick(:status)
  end
```

This will avoid selecting the fields that we're not using. It will also return `nil` if the record is missing.

Reply:
That said, there are a few additional cases to account for:
- Multiple records created, but latest one is not being used (the relationships in `user.rb` permit this)
- Record has a different final state that merits starting over, e.g. `expired`

Reply:
unsure about the 2nd point you mentioned. For this pr we are checking if the status is expired in order to make sure the user starts over, but not sure what other status would warrant that.

Reply:
This PR only checks against `cancelled`, which is a distinct state from `expired`.

Reply:
I think on the second point I may have been distracted by something else & misremembered the check as comparing with `cancelled`. The first point seems like it may still be relevant, though.

Reply:
ok i was trying to figure out how to ensure the most recent enrollment would be chosen and it seems like `pick` is going to take the [1st item from the query results ](https://apidock.com/rails/ActiveRecord/Calculations/pick): `relation.limit(1).[pluck](https://apidock.com/rails/ActiveRecord/Calculations/pluck)(*column_names).first`

Reply:
Yes, in this case the `order(created: :desc)` ensures that the first result will be the most recent enrollment. That should generally be the correct one, though there is a small chance that the user may have an earlier enrollment in a non-final state (though that may be a bug).